:rocket: Helm Chart publish using Github Pages and actions

tiagolobocastro opened the pull request:
Adds new GHA release workflow:

A new release.ylm workflow orchestrates the release process into a number of steps:

1. Push Tag
1.a. check tag and version matches
2. Check it's a new release
3. Build and Push tagged image
4. K8s test using the tagged image
5. If something fails, then we should delete tag, add fixes and retag
6. Publish release and chart to gh-pages
7. Sanity check that the published chart is working

The build has also been modified for different registries and namespaces, meaning you can test this workflow from a fork. You only need to setup your own dockerhub credentials on the fork, or point it to ghcr.io

mhkarimi1383 committed:
:rocket: Helm Chart publish using Github Pages and actions

Signed-off-by: Muhammed Hussein Karimi <info@karimi.dev>

mhkarimi1383 committed:
:pencil2: Chart base path

Signed-off-by: Muhammed Hussein Karimi <info@karimi.dev>

mhkarimi1383 committed:
:green_heart: Change release name template

We are having a single helm chart so we don't need to include chart name

Signed-off-by: Muhammed Hussein Karimi <info@karimi.dev>

mhkarimi1383 committed:
:twisted_rightwards_arrows: Merge develop into helm-publish

Signed-off-by: Muhammed Hussein Karimi <info@karimi.dev>

mhkarimi1383 committed:
:coffin: Remove depricated nix cache action

Signed-off-by: Muhammed Hussein Karimi <info@karimi.dev>

mhkarimi1383 committed:
Merge branch 'develop' into helm-publish

Signed-off-by: Muhammed Hussein karimi <info@karimi.dev>

mhkarimi1383 committed:
:memo: Mention new helm repository in README

Signed-off-by: Muhammed Hussein Karimi <info@karimi.dev>

mhkarimi1383 committed:
:twisted_rightwards_arrows: Merge branch 'develop' into helm-publish

Signed-off-by: Muhammed Hussein Karimi <info@karimi.dev>

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

mhkarimi1383 committed:
:memo: Fix manifests link

Signed-off-by: Muhammed Hussein Karimi <info@karimi.dev>

tiagolobocastro committed:
ci(gha): add new release workflow orchestrator

A new release.ylm workflow orchestrates the release process into a
number of steps:

1. Push Tag
1.a. check tag and version matches
2. Check it's a new release
3. Build and Push tagged image
4. K8s test using the tagged image
5. If something fails, then we should delete tag, add fixes and retag
6. Publish release and chart to gh-pages
7. Sanity check that the published chart is working

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
Merge branch 'develop' into gha-release

tiagolobocastro commented:
@mhkarimi1383 is this a good time to rename the chart from rawfile-csi into rawfile-localpv to match the package name? Also would match the other openebs engines.

Any thoughts @addyess ?

mhkarimi1383 commented:
I completely agree  with this change :)

mhkarimi1383 commented on shell.nix:
```diff
@@ -28,7 +29,9 @@ pkgs.mkShell {
     poetry install
     source $(poetry env info -p)/bin/activate
-    pre-commit install
+    if [ "${CI:-}" = "1" ]; then
```
Is this reverse or something?

tiagolobocastro replied:
doh, yeah that's wrong

tiagolobocastro replied:
Done

tiagolobocastro commented:
> oops sorry, pushed a tag by mistake, already deleted it

tiagolobocastro committed:
ci: don't install pre-commit

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
refactor: rename rawfile-csi to rawfile-localpv

As it says on the tin.
This makes the chart name the same as the package and it's also in the
same format as the other openebs engines.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
docs: a few typos

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
Merge branch 'develop' into gha-release

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
@mhkarimi1383 PTAL, I think this is ready

mhkarimi1383 commented:
@tiagolobocastro 
Nice :)
Approved 💯

tiagolobocastro committed:
Merge branch 'develop' into gha-release
